jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpException.java

```diff
@@ -0,0 +1,128 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import org.eclipse.jetty.io.QuietException;
+import org.eclipse.jetty.util.ExceptionUtil;
+
+/**
+ * <p>A tagging interface for Exceptions that carry a HTTP response code and reason.</p>
+ * <p>Exception sub-classes that implement this interface will be caught by the container
+ * and the {@link #getCode()} used to send a response.</p>
+ */
+@SuppressWarnings("serial")
+public interface HttpException extends QuietException
+{
+    int getCode();
+
+    String getReason();
+
+    static Throwable asThrowable(HttpException httpException)
+    {
+        return httpException instanceof Throwable throwable ? throwable : null;
+    }
+
+    static void rethrowRuntime(HttpException httpException)
```

Comment:
I would prefer this to be `static RuntimeException asRuntimeException(HttpException httpException)` and in the places where it is used have `throw HttpException.asRuntimeException(httpException);`.

Reply:
Can't do that as we'd need a multi-variant return of Error or RuntimeException.
I'll rename to `throwAsUnchecked(HttpException)`

Reply:
Not sure I follow. An `HttpException` cannot be an `Error`, and it is always a `RuntimeException`.

Reply:
@sbordet a HttpException is just a marker interface. Currently we apply it to `RuntimeException` and `IllegalArgumentException`, but in future we may tag an Error as a HttpException.  Unlikely, but having just made a whole PR about decoupling `HttpException` from the specific type of exception, it seams silly to bind it here.